tests/test_association_db_sqlite.py

```diff
@@ -0,0 +1,321 @@
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import unittest
+
+from lsst.ap.association import \
+    AssociationDBSqliteTask, \
+    DIAObjectCollection
+from lsst.afw.coord import Coord
+import lsst.afw.geom as afwGeom
+import lsst.utils.tests
+from test_dia_collection import create_test_dia_objects
+
+
+class TestAssociationDBSqlite(unittest.TestCase):
+
+    def setUp(self):
+        """ Initizlize an empty database.
+        """
+        self.assoc_db = AssociationDBSqliteTask()
+        self.assoc_db.create_tables()
+        self.assoc_db._commit()
+
+    def tearDown(self):
+        """ Close the database connection and delete the object.
+        """
+        self.assoc_db.close()
+        del self.assoc_db
+
+    def test_load(self):
+        """ Test loading of DIAObjects.
+        """
+
+        # Create DIAObjects with two sources per object. Increment their
+        # RA, DEC by 0.04 for each object
+        n_sources_per_object = 2
+        dia_objects = create_test_dia_objects(
+            n_objects=5,
+            n_sources=n_sources_per_object,
+            increment_degrees=0.04,
+            scatter_arcsec=0.0)
+        dia_collection = DIAObjectCollection(dia_objects)
+        self.assoc_db = AssociationDBSqliteTask()
+        self.assoc_db.create_tables()
+
+        self.assoc_db.store(dia_collection, True)
+
+        ctr_point = Coord(
+            afwGeom.Angle(0.08 * afwGeom.degrees),
+            afwGeom.Angle(0.08 * afwGeom.degrees))
+        output_dia_collection = self.assoc_db.load(
+            ctr_point, afwGeom.Angle(0.2))
+
+        for obj_idx in range(5):
+            self.assertEqual(
+                output_dia_collection.dia_objects[obj_idx].n_dia_sources, 2)
+            self._compare_source_records(
+                output_dia_collection.dia_objects[obj_idx].dia_object_record,
+                dia_collection.dia_objects[obj_idx].dia_object_record)
+
+            for record_a, record_b in zip(
+                    output_dia_collection.dia_objects[
+                        obj_idx].dia_source_catalog,
+                    dia_collection.dia_objects[obj_idx].dia_source_catalog):
```

Comment:
This test might actually be too specific: `load` doesn't (and probably shouldn't) guarantee a particular order for the returned objects. Is there an easy way to compare these collections as sets rather than as lists? Or does `dia_source_catalog` sort the sources for you?

Reply:
Agreed. I changed the testing of the DIAObjects to reflect this, looking them up instead by id. I also stor the catalog of DIASources I get by id before comparing them to the SourceCatalog that was input.